src/github.com/getlantern/flashlight/geolookup/geolookup.go

```diff
@@ -25,60 +24,41 @@ var (
 	lookupData  *geolookup.City
 )
 
-func getUserGeolocationData() *geolookup.City {
+func getUserGeolocationData(client *http.Client) (*geolookup.City, error) {
 	lookupMutex.Lock()
 	defer lookupMutex.Unlock()
 
-	var err error
-
 	if lookupData != nil {
 		// We already looked up IP's information.
-		return lookupData
+		return lookupData, nil
 	}
 
-	for {
-		if !geolookup.UsesDefaultHTTPClient() {
-			// Will look up only if we're using a proxy.
-			lookupData, err = geolookup.LookupCity("")
-			if err == nil {
-				// We got what we wanted, no need to query for it again, let's exit.
-				return lookupData
-			}
+	var err error
+	for i := 0; i < maxRetries; i++ {
```

Comment:
I changed this logic a little.

1. Once we have a fronted dialer, we don't need to wait for any proxy to come online.  The only reason it would fail is if there's some sort of error.

2. I capped the number of retries, because as written originally this would hang indefinitely as long as it's failing.